Modules/_decimal/_decimal.c

```diff
@@ -3554,9 +3615,10 @@ dec_as_integer_ratio(PyObject *self, PyObject *args UNUSED)
         return NULL;
     }
 
+    decimal_state *state = GLOBAL_STATE();
```

Comment:
Why not just above line 3621 where it is used first?

Reply:
Is state implicitly needed by the `CURRENT_CONTEXT` macro? If so, that's a bad design; if it is needed, it should be passed as a param, IMO.

Reply:
> Is state implicitly needed by the `CURRENT_CONTEXT` macro? If so, that's a bad design; if it is needed, it should be passed as a param, IMO.

**`CURRENT_CONTEXT` macro doesn't implicitly depend on `state`.** 

The `current_context` function contains global variable that need to be moved to the module state. In the [original PR](https://github.com/python/cpython/pull/103381), I pass the `state` as a parameter to the `CURRENT_CONTEXT` macro. When I broke the PR, I did not adjust the position of the statement. That's why the statement is above the `CURRENT_CONTEXT` macro. :)

I will change the position of this statement to avoid misleading.

Reply:
Thanks; there's a lot of confusing occurrences like this in this PR. Please take care of the other occurrences as well.